src/script/descriptor.cpp

```diff
@@ -883,13 +883,17 @@ class DescriptorImpl : public Descriptor
     virtual bool ToStringSubScriptHelper(const SigningProvider* arg, std::string& ret, const StringType type, const DescriptorCache* cache = nullptr) const
     {
         size_t pos = 0;
+        bool is_private{type == StringType::PRIVATE};
+        bool any_success{!is_private};
```

Comment:
In 23f8b6035ff6538134be3d3df8a3ab8ba47f86c0 _descriptor: ToPrivateString() pass if  at least 1 priv key exists_: the variable name `any_success` had me confused. If you have to retouch, it would be good to document its meaning or rename it.

```cpp
// If a public string is requested, fail if no subscript returns
a public string. If a  private string is requested, fail if no 
subscript returns a private string.
```

Seems like `any_success` was suggested here: https://github.com/bitcoin/bitcoin/pull/32471#discussion_r2342296606

Reply:
I added comments where `any_success` was used.